examples/with-apollo-auth/components/SigninBox.js

```diff
@@ -12,67 +12,70 @@ const SIGN_IN = gql`
 `
 
 // TODO: Find a better name for component.
-const SigninBox = ({ client }) => {
+const SigninBox = () => {
+  const client = useApolloClient()
+
+  const onCompleted = data => {
+    // Store the token in cookie
+    document.cookie = cookie.serialize('token', data.signinUser.token, {
+      maxAge: 30 * 24 * 60 * 60 // 30 days
+    })
+    // Force a reload of all the current queries now that the user is
+    // logged in
+    client.cache.reset().then(() => {
+      redirect({}, '/')
+    })
+  }
+
+  const onError = error => {
+    // If you want to send error to external service?
+    console.error(error)
+  }
+
+  const [signinUser, { error }] = useMutation(SIGN_IN, {
+    onCompleted,
+    onError
+  })
+
   let email, password
 
   return (
-    <Mutation
-      mutation={SIGN_IN}
-      onCompleted={data => {
-        // Store the token in cookie
-        document.cookie = cookie.serialize('token', data.signinUser.token, {
-          maxAge: 30 * 24 * 60 * 60 // 30 days
-        })
-        // Force a reload of all the current queries now that the user is
-        // logged in
-        client.cache.reset().then(() => {
-          redirect({}, '/')
+    <form
+      onSubmit={e => {
+        e.preventDefault()
+        e.stopPropagation()
+
+        signinUser({
+          variables: {
+            email: email.value,
+            password: password.value
+          }
         })
-      }}
-      onError={error => {
-        // If you want to send error to external service?
-        console.log(error)
+
+        email.value = password.value = ''
       }}
     >
-      {(signinUser, { data, error }) => (
-        <form
-          onSubmit={e => {
-            e.preventDefault()
-            e.stopPropagation()
-
-            signinUser({
-              variables: {
-                email: email.value,
-                password: password.value
-              }
-            })
-
-            email.value = password.value = ''
-          }}
-        >
-          {error && <p>No user found with that information.</p>}
-          <input
-            name='email'
-            placeholder='Email'
-            ref={node => {
-              email = node
-            }}
-          />
-          <br />
-          <input
-            name='password'
-            placeholder='Password'
-            ref={node => {
-              password = node
-            }}
-            type='password'
-          />
-          <br />
-          <button>Sign in</button>
-        </form>
-      )}
-    </Mutation>
+      {error && <p>No user found with that information.</p>}
+      <input
+        name='email'
+        placeholder='Email'
+        ref={node => {
+          email = node
+        }}
+      />
+      <br />
+      <input
+        name='password'
+        placeholder='Password'
+        ref={node => {
```

Comment:
It will be great to use useRef() inside React Hooks.

Reply:
That's a good point. Didn't notice it. I will add it as soon as i can